go/prometheusexporter/collector/collector.go

```diff
@@ -201,6 +243,18 @@ func labelsFromLabelPairs(pairs []*promclient.LabelPair) prometheus.Labels {
 	return labels
 }
 
+func (c *Collector) deleteSummaryLabels(summary *prometheus.SummaryVec, labels prometheus.Labels) bool {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+	return summary.Delete(labels)
+}
+
+func (c *Collector) deleteHistogramLabels(histogram *prometheus.HistogramVec, labels prometheus.Labels) bool {
+	c.mutex.Lock()
```

Comment:
Same case as mentioned for the summary. I don't think that this needs to be guarded by the mutex that was introduced to guard the stream config in the collector.

Reply:
histogramvecs might be deleted from the collector state in function Update(). That's why we need to protect the pointer access. But that has happened outside the function. Maybe brain must have been half fried when I wrote this.